### PR TITLE
Feature/loading spinner

### PR DESCRIPTION
**Pull request by Ezeh20**
- This PR holds the loading spinner and the loading message 

Screenshot (80)
Screenshot (79)

### CODE REVIEW

File: src/components/LoadingMessage/internals/constants.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+     Your dreams are valid, regardless of your age.`,
+]
+
+/**
+ * params - {array, max, min = 0}
+ * generate a random message from an array of messages.
+ * A ramdom number will be generated using the range between 0 - { the length of the array }
+ * returns a random message using the position of the message in the array
+ */
+export const randomMessage = (array, max, min = 0) => {
```
Comment (edustipend-root) [inline src/components/LoadingMessage/internals/constants.js:16]: I think we can export this into a separate util that we an use throughout the entire app.
Please create a `src/utils.js` file and move this there.
Also let's add tests for it.

File: src/App.js [hunk 2]
```diff
@@ -5,9 +5,14 @@ import './App.css';
...
 import NotifyModal from './sections/NotifyMe/internals/NotifyModal';
 import NotifyMe from './sections/NotifyMe';
+import Modal from './components/Modal';
+import { ModalContext } from './context/ModalContext';
+import { useContext } from 'react';
+import LoadingMessage from './components/LoadingMessage';
 initFirebaseApp();
 
 function App() {
+  const { notifyLoading } = useContext(ModalContext);
```
Comment (edustipend-root) [inline src/App.js:15]: Can we rename this to `isLoading` so we stay consistent?

File: src/App.js [hunk 3]
```diff
@@ -16,6 +21,11 @@ function App() {
       <NotifyModal>
         <NotifyMe />
       </NotifyModal>
+      {notifyLoading && (
```
Comment (edustipend-root) [inline src/App.js:24]: Can't we rewrite this logic so we don't have to render the modal twice?

So something like

```
 <NotifyModal>
   { isLoading ? <LoadingMessage size='large' /> : <NotifyMe /> }
 </NotifyModal>
```

File: src/components/Button/styles.css [hunk 4]
```diff
@@ -24,10 +28,6 @@
   right: 30px;
 }
 
-.disabled .edustipend-button {
```
Comment (edustipend-root) [inline src/components/Button/styles.css:27]: Why are we removing this?
We want to keep the button disabled when it's the loading state no?

**Review by edustipend-root — CHANGES_REQUESTED** [review]
Good job @Ezeh20, some minor changes I would like that you address.

File: src/components/LoadingMessage/internals/constants.js (hunk 1 above)
Comment (Ezeh20) [reply src/components/LoadingMessage/internals/constants.js:16]: Will do

File: src/App.js (hunk 2 above)
Comment (Ezeh20) [reply src/App.js:15]: Alright

File: src/App.js (hunk 3 above)
Comment (Ezeh20) [reply src/App.js:24]: Tried that, didn't work. The NotifyModal is global so it always pops up, i just made a separate modal for it using the same styles

File: src/components/Button/styles.css (hunk 4 above)
Comment (Ezeh20) [reply src/components/Button/styles.css:27]: The button can still be disabled, that style is not needed

File: src/App.js (hunk 3 above)
Comment (Ezeh20) [reply src/App.js:24]: Cleaned it up a little

**Review by edustipend-root — APPROVED** [review]
Thanks for making the changes @Ezeh20, looks good for now.
Can you confirm your VS code format settings? Your PRs seem to have a lot of formatting changes.

**Pull request merged**